components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -3493,18 +3493,24 @@ public void addRating(APIIdentifier apiIdentifier, int rating, String userId, Co
             String sqlAddQuery;
             String ratingId = UUID.randomUUID().toString();
             if (!userRatingExists) {
-                //This query to update the AM_API_RATINGS table
+                //This query to insert into the AM_API_RATINGS table
                 sqlAddQuery = SQLConstants.ADD_API_RATING_SQL;
+                ps = conn.prepareStatement(sqlAddQuery);
+                ps.setString(1, ratingId);
+                ps.setInt(2, rating);
+                ps.setInt(3, apiId);
+                ps.setInt(4, subscriber.getId());
             } else {
-                //This query to insert into the AM_API_RATINGS table
+                // This query to update the AM_API_RATINGS table
```

Comment:
We should change the method name from addRating to addOrUpdateRating

Reply:
Noted, change has been made. Thanks for pointing out.